### ISSUE TITLE
Mysql adapter should support raw (return after write) in the case where there is only one single raw field of type :id

### ISSUE THREAD

**Issue by leiyangyou** [elixir-lang/ecto#1381]
we ran into a case where the primary key is not of an id type, but we have an auto increment column that we would like to return after insert, the current mysql adapter does not allow for this case.

**Comment by josevalim** [comment]
We need more information on how to reproduce this. Please include full details of your schema, what is your actual primary key and which operations you are trying to perform. If you can include a failing test case to our test suite, even better. Thank you.

**Comment by leiyangyou** [comment]
an example schema looks like the following
```
  @primary_key {:id, Ecto.UUID, autogenerate: false}
  schema "posts" do
    field :short_id, :id, read_after_writes: true
  end
```
where short_id is an auto_increment id column in the db

**Comment by leiyangyou** [comment]
an example (hacky) fix is attached #1382

**Comment by josevalim** [comment]
Closing in favor of the PR!
